rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -980,27 +1002,27 @@ private String endLRA(URL lra, boolean confirm) throws GenericLRAException {
 
             Current.pop(lra);
 
-            URL nextLRA = Current.peek();
+            URI nextLRA = Current.peek();
 
             if (nextLRA != null) {
                 try {
                     init(nextLRA);
-                } catch (URISyntaxException ignore) {
+                } catch (URISyntaxException | MalformedURLException ignore) {
                     // the validity of the url was checked when we added it to Current
                 }
             }
         }
     }
 
-    private void validateURL(String url, boolean nullAllowed, String message) {
+    private void validateURI(URI url, boolean nullAllowed, String message) {
         if (url == null) {
             if (!nullAllowed) {
                 throw new GenericLRAException(null, NOT_ACCEPTABLE.getStatusCode(),
                         String.format(message, "null value"), null);
             }
         } else {
             try {
-                new URL(url);
+                url.toURL(); // ensure it's a valid URI
```

Comment:
does this mean it should be ensured (aka. this is TODO) or the call ensures validity?

Reply:
The former, I added the comment:
// the passed in URI should be a valid URL - verify that that is the case